internal/peer/node/start.go

```diff
@@ -872,6 +873,10 @@ func serve(args []string) error {
 	// Register the Endorser server
 	pb.RegisterEndorserServer(peerServer.Server(), auth)
 
+	// register the snapshot server
+	snapshotSvc := &snapshotgrpc.SnapshotService{LedgerGetter: peerInstance}
+	pb.RegisterSnapshotServer(peerServer.Server(), snapshotSvc)
```

Comment:
Instead of `peerInstance`, it should pass the `LedgerMgr`.  The peerInstance should not be passed to the lower level components.

Reply:
Do you mean to say that we can implement these APIs on the `LedgerMgr` or use the ledger object maintained locally in the `ledgermgmt` pkg while implementing these APIs as a method on `SnapshotService` struct?

Reply:
The latter would again lead to a very clumsy cycle, as there is only one instance of the `LedgerMgr` and peer package will pass that instance back to the `SnapshotService` from the same package. So, I mentioned only the former among the two choices. Further, for the former, though it may bring simplicity of dependencies, hardly makes sense making `LedgerMgr` as a `SnapshotService`. That's the detail of my rational to tending to incline for a separate package.

Reply:
As we discussed offline, it requires a larger refactoring to fix current implementation of `start.go` and how all gRPC services are implemented/called/their entry-point. Hence, it is out of scope for this PR. I agree with your assessment and suggestion.